doc/dune-files.rst

```diff
@@ -285,6 +285,23 @@ language: The syntax is as a list of the following elements:
 
    dep-specification = dep+
 
+.. _always-add-cflags:
+
+future_c_and_cxx_flags_handling
+-----------------
```

Comment:
This heading marker should extend to the length of the title. Otherwise sphinx will complain